internal/model/historical_recommendation_set.go

```diff
@@ -22,11 +26,16 @@ type HistoricalRecommendationSet struct {
 func (r *HistoricalRecommendationSet) CreateHistoricalRecommendationSet() error {
 	db := database.GetDB()
 	result := db.Clauses(clause.OnConflict{
-		Columns:   []clause.Column{{Name: "workload_id"}, {Name: "container_name"}, {Name: "monitoring_end_time"}},
+		Columns:   []clause.Column{{Name: "org_id"}, {Name: "workload_id"}, {Name: "container_name"}, {Name: "monitoring_end_time"}},
 		DoUpdates: clause.AssignmentColumns([]string{"monitoring_start_time", "monitoring_end_time", "recommendations", "updated_at"}),
 	}).Create(r)
 
 	if result.Error != nil {
+		if strings.Contains(result.Error.Error(), "no partition") {
+			partitionMissing.With(prometheus.Labels{"resource_name": "historical_recommendation_set"}).Inc()
+			dbError.Inc()
+			return fmt.Errorf("parition not found for resource_name = %s, org_id = %s, end_time = %s", "workload_metrics", r.OrgId, r.MonitoringEndTime.String())
```

Comment:
Nitpick: 
```suggestion
			return fmt.Errorf("Partition not found for resource %s with org_id %s and end_time %s", "historical_recommendation_set", r.OrgId, r.MonitoringEndTime.String())
```

Reply:
`fmt.Errorf` cannot start with Capital letter, rest I will make the changes.